### PR TITLE
404 Not found page

### LINKED ISSUE #63: 404 Page
Make sure the 404 page is displayed when an unknown route is visited, example:

- `/experiments/2323123`
- `/no-match`

### Design

https://projects.invisionapp.com/d/main#/console/13557531/301870911/preview

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

Close #63 

## Purpose/Implementation Notes

Adds a 404 error page. Catches errors on the Experiments page and the Downloads page and redirects to this page.

The no found route is declared twice in the router, one at the end matching all unknown routes and the other one matching the route `/not-found`. This will allow us to redirect programmatically to this route from the action creators.

https://github.com/AlexsLemonade/refinebio-frontend/blob/e0eca05dd7d689856bfab36c6688272309d65dce/src/components/App/index.js#L32-L33

On the action creators I used `dispatch(replace('/not-found'))`, to navigate to the 404 page and avoid adding an entry on the browser history for this page.

## Types of changes

* [x] New feature (non-breaking change which adds functionality)

## Functional tests

Checked several urls to see that the 404 page is displayed:
- http://localhost:3000/no-match
- http://localhost:3000/experiments/5232323
- http://localhost:3000/download/2bf1c9a5-ce16-4d4a-925e-74974509b025
- http://localhost:3000/dataset/730ad4b9-f789-48a0-a114-ca3a8c5ab030

## Screenshots

image

### CODE REVIEW

File: src/containers/NoMatch/NoMatch.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+  margin-top: 15*8px;
+
+  &__text {
+    padding-top: 7*8px;
+  }
+
+  &__image {
```
Comment (ramenhog) [inline src/containers/NoMatch/NoMatch.scss]: oops? 🙃

File: src/state/viewDownload/actions.js [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
 
 export const loadViewDownload = data => ({
   type: 'LOAD_VIEW_DOWNLOAD',
   data
 });
 
 export const fetchDataSetDetailsForView = dataSetId => async dispatch => {
-  // 1. fetch dataset information
-  const { data: dataSet } = await getDataSet(dataSetId);
+  console.log(dataSetId);
```
Comment (ramenhog) [inline src/state/viewDownload/actions.js]: 🗑

File: src/state/viewDownload/actions.js [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
+    const { experiments, samples } = await getSamplesAndExperiments(dataSet);
 
-  // 3. store data in `viewDownload` reducer
-  dispatch(loadViewDownload({ dataSet, experiments, samples }));
+    // 3. store data in `viewDownload` reducer
+    dispatch(loadViewDownload({ dataSet, experiments, samples }));
+  } catch (e) {
+    // TODO: check the type of error
+    dispatch(replace('/no-match'));
+    return;
```
Comment (ramenhog) [inline src/state/viewDownload/actions.js]: we probably don't need this `return`, right?

**Comment by arielsvn** [comment]
Thanks for reviewing @ramenhog ! I addressed the comments :)

**Pull request merged**